### PR TITLE
Add queries to migrate order, line and transaction from MicroMaster

### PR DESCRIPTION
**Pull request by rachellougee**
#### Pre-Flight checklist
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/657

#### What's this PR do?

- Adding a new state 'partially_refunded' to bring over partially refunded orders from MM
- Making 'reference_number' as readly-only field in django admin (This field can't be manually modified as we rely on it for migration)
- Adding migration queries to bring over orders/lines/transactions from MicroMaster

#### How should this be manually tested?

- Run the following query to check if there are corresponding products for the course runs in the local DB
   (If any course run doesn't have a product, then add them from Django admin)
  
```
     SELECT * FROM courses_courserun AS run 
     LEFT JOIN public.ecommerce_product AS product 
       ON run.id = product.object_id AND product.content_type_id = 60
    WHERE product.id is NULL
```
 - Create a local foreign data connection named 'micromasters'  (Only need this step when testing locally)
 ```
CREATE EXTENSION IF NOT EXISTS postgres_fdw;

CREATE SERVER IF NOT EXISTS micromaster_bridge
	FOREIGN DATA WRAPPER postgres_fdw
    OPTIONS (host 'micromasters-rc-db.cbnm7ajau6mi.us-east-1.rds.amazonaws.com', dbname 'micromasters', port '15432');

CREATE USER MAPPING IF NOT EXISTS FOR postgres
	SERVER micromaster_bridge
    OPTIONS (user 'micromasters', password '------');

DROP SCHEMA IF EXISTS micromasters CASCADE;
CREATE SCHEMA micromasters;
IMPORT FOREIGN SCHEMA public 
FROM SERVER micromaster_bridge INTO micromasters;
```

- Run the migration script
```
docker-compose run --rm web ./manage.py import_micromasters_data
```

- Verify orders/line/transactions are imported correctly (compare with MicroMaster RC using reference_number)

Orders
   
image

Line:

image

Transaction
image

Receipt on frontend
image

### CODE REVIEW

**Comment by rachellougee** [comment]
@rhysyngsun This is ready for review as the blocker is merged. Thank you

File: micromasters_import/management/commands/queries/007_import_transation.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    updated_on
+)
+SELECT
+    mm_receipt.data->>'transaction_id',
+    mm_order.total_price_paid,
+    mm_receipt.data,
+    mo_order.id,
+    CASE
+         WHEN mm_order.status IN ('refunded', 'partially_refunded') THEN 'refund'
+         ELSE 'payment'
```
Comment (rhysyngsun) [inline micromasters_import/management/commands/queries/007_import_transation.sql:18]: You added the partially refunded state, so should this be mapped to that?

**Review by rhysyngsun — CHANGES_REQUESTED** [review]

File: micromasters_import/management/commands/queries/007_import_transation.sql (hunk 1 above)
Comment (rachellougee) [reply micromasters_import/management/commands/queries/007_import_transation.sql:18]: This is different from the order state. This is the transaction state that only has two:
TRANSACTION_TYPE_REFUND = "refund"
TRANSACTION_TYPE_PAYMENT = "payment"
I thought the transaction type is just a way to identify if it's a refund or payment.  Should we add a third transaction type?

Comment (rhysyngsun) [reply micromasters_import/management/commands/queries/007_import_transation.sql:18]: Oh, no you're right. I need more coffee. Carry on.

**Review by rhysyngsun — APPROVED** [review]
LGTM

**Pull request merged**